### DISCUSSION TITLE
Fix minimum flow thru a connection

### DISCUSSION THREAD

**Discussion by emelinemalaussene** [Spine-project/Spine-Toolbox#2040] [Support]
Hi! We are two students working with Spine-Toolbox to simulate hydropower in a river. We want to model a constraint on the spill from a hydropower plant, so we want a minimum flow thru the connection between two nodes. Is this possible to do with a connection, connection__from_node, connection__to_node or connection__node__node parameter? We understand that we can set a min. ratio between the in and out flow, but in our case we have a fix value on the minimum flow that we want to use.

To illustrare what we want to do: we would like to be able to set a fix minimum flow in the Rebnis_to_Bergnäs_spill connection.
<img width="428" alt="Skärmavbild 2023-03-17 kl  10 48 38" src="https://user-images.githubusercontent.com/128138048/225870614-cb847fa3-69fd-4f1c-ad93-0427f4057c51.png">

Has anyone done something similar or has an idea on how we could solve this problem?
Thank you in advance!

**Comment by manuelma** [comment]
It looks like we don't have min_connection_flow? We just have min_unit_flow... That's annoying and I think we should implement it. Otherwise the only way to achieve what @emelinemalaussene is asking is to insert a dummy unit plus node in between so min_unit_flow can be used - but that sounds terrible!

**Comment by DillonJ** [comment]
You can also use user constraints to accomplish this but I agree with @manuelma that it should be supported more directly. In fact I thought we were going to do this at the same time as the first initial_[variable] update

**Comment by manuelma** [comment]
I didn't notice min_connection_flow was missing when I did the initial_[variable] thing - otherwise I think I would have implemented it. It is very easy to do but also a little time consuming. It would be great to have a new collaborator to do it.

**Comment by emelinemalaussene** [comment]
Thank you for the help! 
We solved the problem by adding the minimum flow to the demand in the upper node and then subtracted it from the demand in the next upper node. We had another idea, using almost the same concept, where we would create a parallel connection to the Rebnis_to_Bergnäs_spill where we could use the fix_connection_flow to simulate a minimum flow.
